src/main/java/sparksoniq/spark/iterator/SparkRuntimeIterator.java

```diff
@@ -70,6 +70,10 @@ public Item next(){
                 result = _rdd.collect();
         }
 
+        if(result.size() == 0){
```

Comment:
Likewise, this and the above initialization should go to open().

Reply:
@ghislainfourny  that will cause the RDD computation to be executed before next() is actually called. Do we want this?